newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -57,214 +90,180 @@ const themeStyles = theme => ({
+  return optionList;
+};
 
-  forceInputValueTo(newValue: string): void {
-    if (this.state.inputValue !== null) {
-      this.setState({
-        inputValue: newValue,
-      });
+const handleChange = (
+  event: SyntheticKeyboardEvent<HTMLInputElement>,
+  option: Option | string,
+  props: Props,
+  ref: { current: ?HTMLInputElement }
+): void => {
+  if (option !== null && option.type !== 'separator') {
+    if (typeof option === 'string') props.onChange(option);
+    else {
+      if (option.onClick) option.onClick();
+      else
+        props.onChoose
+          ? props.onChoose(option.value)
+          : props.onChange(option.value);
     }
   }
+  if (event.key === 'Enter' && ref.current) ref.current.blur();
+};
 
-  isOptionDisabled = (option: Option): boolean => {
-    if (option.type === 'separator') return true;
-    return false;
-  };
-
-  renderItem = (option: Option, state: Object): React.Node => {
-    if (option.type && option.type === 'separator') {
-      return (
-        <ListItem
-          divider
-          disableGutters
-          component={'div'}
-          style={styles.listItem}
-        />
-      );
-    }
-    return (
-      <ListItem component={'div'} style={styles.listItem}>
-        {option.renderIcon && (
-          <ListItemIcon>{option.renderIcon()}</ListItemIcon>
-        )}
-        {option.value}
-      </ListItem>
-    );
+const getDefaultStylingProps = (
+  params: Object,
+  value: string,
+  props: Props,
+  ref: { current: ?HTMLInputElement }
+): Object => {
+  const { InputProps, inputProps, InputLabelProps, ...other } = params;
+  return {
+    ...other,
+    InputProps: {
+      ...InputProps,
+      className: null,
+      endAdornment: null,
+    },
+    inputProps: {
+      ...inputProps,
+      className: null,
+      onKeyDown: (event: SyntheticKeyboardEvent<HTMLInputElement>): void => {
+        if (event.key === 'Enter') handleChange(event, value, props, ref);
+      },
+    },
   };
+};
 
-  render() {
-    const { classes } = this.props;
-
-    const getCurrentInputValue = (): string =>
-      this.state.inputValue !== null ? this.state.inputValue : this.props.value;
-
-    const filterFunction = (options: DataSource, state: Object): DataSource => {
-      const lowercaseInputValue = getCurrentInputValue().toLowerCase();
-      const optionList = options.filter(option => {
-        if (option.type === 'separator') return true;
-        if (!option.text) return true;
-        return option.text.toLowerCase().indexOf(lowercaseInputValue) !== -1;
-      });
+const getOptionLabel = (option: Option, value: string): string =>
+  option.value ? option.value : value;
 
-      if (
-        !optionList.filter(
-          option => option.type !== 'separator' && option.value
-        ).length
-      )
-        return [];
+export default React.forwardRef<Props, Interface>(
+  function SemiControlledAutoComplete(props: Props, ref) {
+    const _input = React.useRef((null: ?HTMLInputElement));
```

Comment:
nitpicking: you can name this `input`, it's a local variable so no risk of external code accessing it, so no need to use _ to show that it's private :)